### PR TITLE
DOCKER-71 fix wrong condition for sonarSecretKey in helm-chart

### PR DESCRIPTION
**Pull request by jCOTINEAU**
- fix condition in helm templating for sonar.properties ConfigMap
- fix condition in helm templating for volumeMounts of this sonar.properties data

tests:
- templating locally with various .values file
- ct tests
- golden test on [that branch](https://github.com/SonarSource/helm-chart-sonarqube/tree/staging/jco/terratest-helm-chart)
- integration test by hand with various .values files

Please ensure your pull request adheres to the following guidelines:
- [x] explain your motives to contribute this change: what problem you are trying to fix, what improvement you are trying to make
- [x] Document your Changes in the `CHANGELOG.md` file of the respected chart as well as the `Chart.yaml`
- [x] Bump the Version number of the respected chart

### CODE REVIEW

File: charts/sonarqube-dce/templates/config.yaml [hunk 1]
```diff
@@ -32,15 +29,12 @@ metadata:
...
-  {{ range $key, $val := .Values.searchNodes.sonarProperties }}
+  {{- range $key, $val := .Values.searchNodes.sonarProperties }}
     {{ $key }}={{ $val }}
-  {{ end }}
   {{- end }}
-    {{- if and .Values.sonarSecretKey .Values.searchNodes.sonarProperties }}
-      sonar.secretKeyPath={{ .Values.sonarqubeFolder }}/secret/sonar-secret.txt
-    {{- end }}
\ No newline at end of file
+  {{- if and .Values.sonarSecretKey }}
```
Comment (leo-geoffroy-sonarsource) [inline charts/sonarqube-dce/templates/config.yaml:37]: I think we don't need the "and" here?

File: charts/sonarqube/templates/config.yaml [hunk 2]
```diff
@@ -8,18 +8,15 @@ metadata:
...
-  {{ range $key, $val := .Values.sonarProperties }}
+  {{- range $key, $val := .Values.sonarProperties }}
     {{ $key }}={{ $val }}
-  {{ end }}
+  {{- end }}
   {{- if not .Values.elasticsearch.bootstrapChecks }}
     sonar.es.bootstrap.checks.disable=true
   {{- end }}
+  {{- if .Values.sonarSecretKey }}
+    sonar.secretKeyPath={{ .Values.sonarqubeFolder }}/secret/sonar-secret.txt
```
Comment (carminevassallo) [inline charts/sonarqube/templates/config.yaml:20]: The path to the encryption key can be specified among the `sonar.properties` (see the [doc](https://docs.sonarqube.org/latest/instance-administration/security/#:~:text=A%20unique%20secret%20key%20must,click%20on%20Generate%20Secret%20Key.&text=Restrict%20file%20permissions%20to%20the,and%20read%2Daccess%20only))).

We should make it sure that the path is not static and gets updated if the `sonar.secretKeyPath` property is set among the sonar.properties. We should probably specify this among the configurable parameters in our helm chart. To sum up: if sonar.secretKeyPath is set among the sonar.properties, we should adjust the path here.

File: charts/sonarqube/CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,9 @@
 # SonarQube Chart Changelog
 All changes to this chart will be documented in this file.
 
+## [5.4.1]
+* Refactor templating of ConfigMap for sonar.properties
```
Comment (carminevassallo) [inline charts/sonarqube/CHANGELOG.md:5]: I would be a little bit more explicit here, saying that we also fix the setting of sonarSecretKey without being strictly dependent on the sonar.properties parameter.

File: charts/sonarqube-dce/CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,9 @@
 # SonarQube Chart Changelog
 All changes to this chart will be documented in this file.
 
+## [4.3.1]
+* Refactor templating of ConfigMap for sonar.properties
```
Comment (carminevassallo) [inline charts/sonarqube-dce/CHANGELOG.md:5]: Same comment as in the standard chart

File: charts/sonarqube-dce/templates/config.yaml [hunk 5]
```diff
@@ -9,18 +9,15 @@ metadata:
...
-  {{ range $key, $val := .Values.ApplicationNodes.sonarProperties }}
+  {{- range $key, $val := .Values.ApplicationNodes.sonarProperties }}
     {{ $key }}={{ $val }}
-  {{ end }}
   {{- end }}
-    {{- if and .Values.sonarSecretKey .Values.ApplicationNodes.sonarProperties }}
-      sonar.secretKeyPath={{ .Values.sonarqubeFolder }}/secret/sonar-secret.txt
-    {{- end }}
+  {{- if .Values.sonarSecretKey }}
+    sonar.secretKeyPath={{ .Values.sonarqubeFolder }}/secret/sonar-secret.txt
```
Comment (carminevassallo) [inline charts/sonarqube-dce/templates/config.yaml:18]: same comment as in the standard chart

File: charts/sonarqube/templates/config.yaml [hunk 6]
```diff
@@ -8,18 +8,15 @@ metadata:
     release: {{ .Release.Name }}
     heritage: {{ .Release.Service }}
 data:
-  {{- if and .Values.sonarSecretKey (not .Values.sonarProperties) (not .Values.elasticsearch.bootstrapChecks) }}
-  sonar.properties: sonar.secretKeyPath={{ .Values.sonarqubeFolder }}/secret/sonar-secret.txt
-  {{- end }}
-  {{- if or .Values.sonarProperties (not .Values.elasticsearch.bootstrapChecks) }}
+{{- if or .Values.sonarProperties .Values.sonarSecretKey (not .Values.elasticsearch.bootstrapChecks ) }}
```
Comment (carminevassallo) [inline charts/sonarqube/templates/config.yaml:11]: This condition is superfluous for me. In the deployment and stateful set, we only get data from this configmap if this condition is met. So, we do a double check that is not needed. However, for readability purposes, I see the opportunity of leaving this condition here, as it improves our understanding of the configmap (maybe a third opinion from @leo-geoffroy-sonarsource is also needed here 😝).

File: charts/sonarqube-dce/templates/config.yaml [hunk 7]
```diff
@@ -9,18 +9,15 @@ metadata:
     release: {{ .Release.Name }}
     heritage: {{ .Release.Service }}
 data:
-  {{- if and .Values.sonarSecretKey (not .Values.ApplicationNodes.sonarProperties) }}
-  sonar.properties: sonar.secretKeyPath={{ .Values.sonarqubeFolder }}/secret/sonar-secret.txt
-  {{- end }}
-  {{- if or .Values.ApplicationNodes.sonarProperties }}
+{{- if or .Values.sonarSecretKey .Values.ApplicationNodes.sonarProperties }}
```
Comment (carminevassallo) [inline charts/sonarqube-dce/templates/config.yaml:12]: similar comment as in the standard chart. Superfluous condition that might stay for readability purposes.

File: charts/sonarqube-dce/templates/config.yaml [hunk 8]
```diff
@@ -32,15 +29,12 @@ metadata:
...
+  {{- range $key, $val := .Values.searchNodes.sonarProperties }}
     {{ $key }}={{ $val }}
-  {{ end }}
   {{- end }}
-    {{- if and .Values.sonarSecretKey .Values.searchNodes.sonarProperties }}
-      sonar.secretKeyPath={{ .Values.sonarqubeFolder }}/secret/sonar-secret.txt
-    {{- end }}
\ No newline at end of file
+  {{- if .Values.sonarSecretKey }}
+    sonar.secretKeyPath={{ .Values.sonarqubeFolder }}/secret/sonar-secret.txt
```
Comment (carminevassallo) [inline charts/sonarqube-dce/templates/config.yaml:38]: same comment as in the standard chart

File: charts/sonarqube-dce/templates/config.yaml [hunk 9]
```diff
@@ -32,15 +29,12 @@ metadata:
     release: {{ .Release.Name }}
     heritage: {{ .Release.Service }}
 data:
-  {{- if and .Values.sonarSecretKey (not .Values.searchNodes.sonarProperties) }}
-  sonar.properties: sonar.secretKeyPath={{ .Values.sonarqubeFolder }}/secret/sonar-secret.txt
-  {{- end }}
-  {{- if or .Values.searchNodes.sonarProperties }}
+{{- if or .Values.sonarSecretKey .Values.searchNodes.sonarProperties }}
```
Comment (carminevassallo) [inline charts/sonarqube-dce/templates/config.yaml:32]: similar comment as in the standard chart. Superfluous condition that might stay for readability purposes.

**Review by carminevassallo — CHANGES_REQUESTED** [review]
Hi @jCOTINEAU, this is a great and big refactoring, thanks for it! We might want to consider the `secretKeyPath` specified among the properties, please have a look at my comments :)

File: charts/sonarqube/templates/config.yaml (hunk 2 above)
Comment (jCOTINEAU) [reply charts/sonarqube/templates/config.yaml:19]: That's a good catch, but we will probably not embed that in this PR, as it is a little bit out of the scope of the initial work.

**Review by carminevassallo — APPROVED** [review]

**Review by leo-geoffroy-sonarsource — APPROVED** [review]

**Pull request merged**